internal/server/auth/server.go

```diff
@@ -83,6 +132,23 @@ func (s *Server) ListAuthentications(ctx context.Context, r *auth.ListAuthentica
 func (s *Server) DeleteAuthentication(ctx context.Context, req *auth.DeleteAuthenticationRequest) (*emptypb.Empty, error) {
 	s.logger.Debug("DeleteAuthentication", zap.String("id", req.Id))
 
+	if s.enableAuditLogging {
+		actor := ActorFromContext(ctx)
+
+		a, err := s.GetAuthentication(ctx, &auth.GetAuthenticationRequest{
+			Id: req.Id,
+		})
+		if err != nil {
+			s.logger.Error("failed to get authentication for audit events", zap.Error(err))
+			return nil, err
+		}
+		if a.Method == auth.Method_METHOD_TOKEN {
+			event := audit.NewEvent(audit.Metadata{Type: audit.Token, Action: audit.Delete, Actor: actor}, a.Metadata)
+			span := trace.SpanFromContext(ctx)
+			span.AddEvent("event", trace.WithAttributes(event.DecodeToAttributes()...))
```

Comment:
This could be done in another PR:
We could define a little `event.AddToContextSpan(ctx)` method.
Just to make this a little more terse.

Reply:
@GeorgeMac Agreed. I can do that.